control_plane/src/endpoint.rs

```diff
@@ -1021,8 +1022,22 @@ impl Endpoint {
         }
     }
 
-    pub fn stop(&self, mode: &str, destroy: bool) -> Result<()> {
-        self.pg_ctl(&["-m", mode, "stop"], &None)?;
+    pub async fn stop(&self, mode: &str, destroy: bool) -> Result<Lsn> {
+        // In production, control plane uses /terminate for shutting down endpoint
+        // which returns LSN after syncing safekeepers.
+        let response = reqwest::Client::new()
+            .post(format!(
+                "http://{}:{}/terminate",
+                self.external_http_address.ip(),
+                self.external_http_address.port()
+            ))
+            .bearer_auth(self.generate_jwt(Some(ComputeClaimsScope::Admin))?)
+            .send()
+            .await?;
+        let lsn: Lsn = serde_json::from_str(&response.text().await?)?;
+
+        // In case /terminate doesn't exit soon
+        let _ = self.pg_ctl(&["-m", mode, "stop"], &None);
```

Comment:
I wonder what is the difference between `mode == "immediate"` and `destroy` set to `true`? Seems like a repetition to me, or otherwise, sounds like another candidate to join the aforementioned suggestion of creating a `StopMode` enum:

```rust
enum StopMode {
    Destroy,
    Immediate,
    Fast,
}

impl StopMode {
    fn should_stop_gracefully(&self) -> bool {
        matches!(self, Self::Immediate | Self::Destroy)
    }

    fn should_cleanup(&self) -> bool {
        *self == Self::Destroy
    }
}
```

Reply:
`pg_ctl` won't be used. As end goal, `neon_local` should use same API as tests